bestbook/api/books.py

```diff
@@ -70,6 +70,7 @@ class Book(core.Base):
     @staticmethod
     def clean_olid(olid):
         """Extract just the olid from some olid-containing string"""
+        olid = olid or ''
         if olid.lower().startswith('ol') and olid.lower()[-1] in ['M', 'W']:
             return olid
         return re.findall(r'OL[0-9]+[MW]', olid)[0]
```

Comment:
I think that this will throw an IndexError if `olid` is an empty string.

Reply:
Good catch.  Fixed.